src/components/ha-dialog.ts

```diff
@@ -63,16 +63,23 @@ export class HaDialog extends MwcDialog {
           min-height: var(--mdc-dialog-min-height, auto);
         }
         .header_button {
+          color: var(--text-primary-color);
           position: absolute;
           right: 16px;
           top: 10px;
           text-decoration: none;
-          color: inherit;
+        }
+        .header_title {
+          margin-right: 40px;
+          color: var(--text-primary-color);
```

Comment:
Why is this needed? We have `--mdc-dialog-heading-ink-color: var(--primary-text-color);
    --mdc-dialog-content-ink-color: var(--primary-text-color);`

Probably the dialog that is not displayed correctly doesn't use `haStyleDialog`